react-native/react/engine/index.js

```diff
@@ -45,14 +45,22 @@ class Engine {
     this.serverListeners = {}
 
     // A list of functions to call when we connect
-    this.onConnectFns = []
+    this.onConnectFns = {}
   }
 
   onConnect () {
-    this.onConnectFns.forEach(f => f())
+    Object.keys(this.onConnectFns).forEach(k => this.onConnectFns[k]())
   }
 
-  listenOnConnect (f) {
+  // Listen to when the engine connects. Will also be called on reconnects.
+  // Multiple calls of the same key are idempotent
+  // The key should be a helpful string.
+  listenOnConnect (key, f) {
+    // We've already registered this key, do nothing
+    if (this.onConnectFns[key] !== undefined) {
```

Comment:
I'm thinking we wanna do the opposite here. likely we want to blow away the old thing and keep the new thing (hot reloaded code, etc)